core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/editor/toolbar/withEditorToolbar.ts

```diff
@@ -0,0 +1,199 @@
+    createPageApiUrlWithQueryParams,
+    createPureURL,
+    getIsDefaultVariant,
+    sanitizeURL
+} from '../../../../utils';
+import { UVEState } from '../../../models';
+import { EditorToolbarState, ToolbarProps } from '../models';
+
+const initialState: EditorToolbarState = {
+    device: null,
+    socialMedia: null,
+    isEditState: true
+};
+
+/**
+ * Add computed properties and methods to the store to handle the Editor Toolbar UI
+ *
+ * @export
+ * @return {*}
+ */
+export function withEditorToolbar() {
+    return signalStoreFeature(
+        {
+            state: type<UVEState>()
+        },
+        withState<EditorToolbarState>(initialState),
+        withComputed((store) => ({
+            $toolbarProps: computed<ToolbarProps>(() => {
+                const params = store.params();
+                const url = sanitizeURL(params.url);
+
+                const pageAPIQueryParams = createPageApiUrlWithQueryParams(url, params);
+                const pageAPIResponse = store.pageAPIResponse();
+                const experiment = store.experiment?.();
+
+                const pageAPI = `/api/v1/page/${
+                    store.isTraditionalPage() ? 'render' : 'json'
+                }/${pageAPIQueryParams}`;
+
+                const isExperimentRunning = experiment?.status === DotExperimentStatus.RUNNING;
+                const shouldShowUnlock =
+                    pageAPIResponse.page.locked && pageAPIResponse.page.canLock;
+
+                const unlockButton = {
+                    inode: pageAPIResponse.page.inode,
+                    loading: store.status() === UVE_STATUS.LOADING
+                };
+                const shoudlShowInfoDisplay =
+                    !store.canEditPage() ||
+                    pageAPIResponse.page.locked ||
+                    !!store.device() ||
+                    !!store.socialMedia();
+
+                const bookmarksUrl = createFavoritePagesURL({
+                    languageId: Number(params.language_id),
+                    pageURI: url,
+                    siteId: pageAPIResponse.site.identifier
+                });
+
+                return {
+                    bookmarksUrl,
+                    copyUrl: createPureURL(params),
+                    apiUrl: `${window.location.origin}${pageAPI}`,
+                    currentLanguage: pageAPIResponse.viewAs.language,
+                    urlContentMap: store.isEditState()
+                        ? (pageAPIResponse.urlContentMap ?? null)
+                        : null,
+                    runningExperiment: isExperimentRunning ? experiment : null,
+                    workflowActionsInode: store.canEditPage() ? pageAPIResponse.page.inode : null,
+                    unlockButton: shouldShowUnlock ? unlockButton : null,
+                    showInfoDisplay: shoudlShowInfoDisplay,
+                    deviceSelector: {
+                        apiLink: `${params.clientHost ?? window.location.origin}${pageAPI}`,
```

Comment:
Should we create a property for this `${params.clientHost ?? window.location.origin}`? It can be a `computed` called `origin`

Reply:
we cannot access computed properties inside the store. Computed properties are a reflection of the UI.